examples/flutter_gallery/lib/demo/cupertino/cupertino_refresh_demo.dart

```diff
@@ -0,0 +1,225 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+import 'dart:math' show Random;
+
+import 'package:flutter/cupertino.dart';
+
+class CupertinoRefreshControlDemo extends StatefulWidget {
+  static const String routeName = '/cupertino/refresh';
+
+  @override
+  _CupertinoRefreshControlDemoState createState() => new _CupertinoRefreshControlDemoState();
+}
+
+class _CupertinoRefreshControlDemoState extends State<CupertinoRefreshControlDemo> {
+  List<List<String>> randomizedContacts;
+
+  @override
+  void initState() {
+    super.initState();
+    repopulateList();
+  }
+
+  void repopulateList() {
+    final Random random = new Random();
+    randomizedContacts = new List<List<String>>.generate(
+      100,
+      (int index) {
+        return contacts[random.nextInt(contacts.length)]
+            // Randomly adds a telephone icon next to the contact or not.
+            ..add(random.nextBool().toString());
+      }
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new DefaultTextStyle(
+      style: const TextStyle(
+        fontFamily: '.SF UI Text',
+        inherit: false,
+        fontSize: 17.0,
+        color: CupertinoColors.black,
+      ),
+      child: new CupertinoPageScaffold(
+        child: new DecoratedBox(
+          decoration: const BoxDecoration(color: const Color(0xFFEFEFF4)),
+          child: new CustomScrollView(
+            slivers: <Widget>[
+              const CupertinoSliverNavigationBar(
+                largeTitle: const Text('Cupertino Refresh'),
+              ),
+              new CupertinoRefreshControl(
+                onRefresh: () {
+                  return new Future<void>.delayed(const Duration(seconds: 2))
+                      ..then((_) => setState(() => repopulateList()));
+                },
+              ),
+              new SliverSafeArea(
+                top: false, // Top safe area is consumed by the navigation bar.
+                sliver: new SliverList(
+                  delegate: new SliverChildBuilderDelegate(
+                    (BuildContext context, int index) {
+                      return new _ListItem(
+                        name: randomizedContacts[index][0],
+                        place: randomizedContacts[index][1],
+                        date: randomizedContacts[index][2],
+                        called: randomizedContacts[index][3] == 'true',
+                      );
+                    },
+                    childCount: 20,
+                  ),
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+List<List<String>> contacts = <List<String>>[
```

Comment:
A list of U.S. presidents is a little U.S.-centric.  Maybe a list of inventors or Nobel prize winners or something more international?

Reply:
ah ya, that's a good idea. I'll try to find another list in a separate PR. Just need to find something with specific dates so it looks similar to the native contacts app.

Reply:
Or just pick a list of presidents or something from another country. :-)